src/shim/hwctx.cpp

```diff
@@ -244,14 +244,30 @@ delete_ctx_on_device()
   fini_log_buf();
 }
 
+void
+hw_ctx::
+set_metadata()
+{
+  m_metadata.magic_no = 0xBAADBEEF;
+  m_metadata.major = 1;
+  m_metadata.minor = 4;
+  m_metadata.num_cols = m_num_cols;
+  for (int i = 0; i < m_num_cols; i++) {
+    m_metadata.col_offset[i] = 1024 * i;
```

Comment:
I'd think both the num_cols and size can be the arguments, thus the code can be a little dynamic here. for example:
set_metadata(num_cols, size) {
...
for (... i < num_cols; ...)
   col_size[i] = size;
   col_offset[i] = size * i;
...
}